### PR TITLE
edited code, variable used before instantiated

### PR DESCRIPTION
**Pull request by GeorgeSavvidis**
<details>
<summary><strong>Instructions</strong></summary>

Thanks for contributing! :heart:

If this contribution is for instructor training, please email the link to this contribution to
checkout@carpentries.org so we can record your progress. You've completed your contribution
step for instructor checkout by submitting this contribution!

Keep in mind that **lesson maintainers are volunteers** and it may take them some time to
respond to your contribution. Although not all contributions can be incorporated into the lesson
materials, we appreciate your time and effort to improve the curriculum. If you have any questions
about the lesson maintenance process or would like to volunteer your time as a contribution
reviewer, please contact The Carpentries Team at team@carpentries.org.

You may delete these instructions from your comment.

\- The Carpentries
</details>

### CODE REVIEW

**Comment by GeorgeSavvidis** [comment]
I have added a call out box. Please let me know if you like how it looks

File: _episodes/03-matplotlib.md [hunk 1]
```diff
@@ -13,15 +13,28 @@ keypoints:
...
 the best way to develop insight is often to visualize data.  Visualization deserves an entire
 lecture of its own, but we can explore a few features of Python's `matplotlib` library here.  While
 there is no official plotting library, `matplotlib` is the _de facto_ standard.  First, we will
-import the `pyplot` module from `matplotlib` and use two of its functions to create and display a
+import the `pyplot` module from `matplotlib`, and then we will use two of its functions to create and display a
 [heat map]({{ page.root }}/reference.html#heat-map) of our data:
 
+> ## Important Note
+>
+> In the previous lesson we set the `data` variable and imported `numpy`. If you have not completed lesson `2. Analyzing Patient Data`,
```
Comment (maxim-belkin) [inline _episodes/03-matplotlib.md:25]: this entire repository is a lesson and it's made of individual episodes. So, we're talking about the previous episode. Let's update the wording and link to the previous episode. The proper way to link to an episode is described in the [lesson-example](https://carpentries.github.io/lesson-example/04-formatting/index.html#locations-and-names), and in our case it'll be
```md
[Analyzing Patient Data]({{ page.root }}{% link _episodes/02-numpy.md %})
```

@ldko, could you please comment on the wording here (if you have any suggestions how to improve this)?

File: _episodes/03-matplotlib.md [hunk 2]
```diff
@@ -13,15 +13,28 @@ keypoints:
...
+> please run the following commands prior to continuing with the current lesson.
+>
+> ~~~
+> import numpy
+> data = numpy.loadtxt(fname='inflammation-01.csv', delimiter=',')
+> ~~~
+> {: .language-python}
+{: .callout}
```
Comment (maxim-belkin) [inline _episodes/03-matplotlib.md:33]: `.callout` is fine but I wonder if it should be a `.prereq`. What do you think, @ldko?

File: _episodes/03-matplotlib.md [hunk 3]
```diff
@@ -13,15 +13,28 @@ keypoints:
 ---
 
 ## Visualizing data
-The mathematician Richard Hamming once said, "The purpose of computing is insight, not numbers," and
+The mathematician Richard Hamming once said, "The purpose of computing is insight, not numbers", and
```
Comment (maxim-belkin) [inline _episodes/03-matplotlib.md:16]: Please undo this change.
https://www.grammar.com/Commas-with-Quotation-Marks

File: _episodes/03-matplotlib.md [hunk 4]
```diff
@@ -13,15 +13,28 @@ keypoints:
...
+> ~~~
+> import numpy
+> data = numpy.loadtxt(fname='inflammation-01.csv', delimiter=',')
+> ~~~
+> {: .language-python}
+{: .callout}
+
 ~~~
 import matplotlib.pyplot
+data = numpy.loadtxt(fname='inflammation-01.csv', delimiter=',')
```
Comment (maxim-belkin) [inline _episodes/03-matplotlib.md:37]: Please remove this line

File: _episodes/03-matplotlib.md [hunk 5]
```diff
@@ -62,7 +75,7 @@ matplotlib.pyplot.show()
 
 ![Minimum Value Along The First Axis](../fig/inflammation-01-minimum.svg)
 
-The maximum value rises and falls smoothly, while the minimum seems to be a step function.  Neither
+The maximum value rises and falls smoothly, while the minimum values seems to be a step function.  Neither
```
Comment (maxim-belkin) [inline _episodes/03-matplotlib.md:78]: @ldko, could you please comment on this change?

File: _episodes/03-matplotlib.md [hunk 6]
```diff
@@ -13,15 +13,28 @@ keypoints:
 ---
 
 ## Visualizing data
-The mathematician Richard Hamming once said, "The purpose of computing is insight, not numbers," and
+The mathematician Richard Hamming once said, "The purpose of computing is insight, not numbers", and
 the best way to develop insight is often to visualize data.  Visualization deserves an entire
 lecture of its own, but we can explore a few features of Python's `matplotlib` library here.  While
 there is no official plotting library, `matplotlib` is the _de facto_ standard.  First, we will
-import the `pyplot` module from `matplotlib` and use two of its functions to create and display a
+import the `pyplot` module from `matplotlib`, and then we will use two of its functions to create and display a
```
Comment (maxim-belkin) [inline _episodes/03-matplotlib.md:20]: @ldko, could you please comment on this change?

**Review by maxim-belkin — CHANGES_REQUESTED** [review]
Thanks for the PR, @GeorgeSavvidis!
I requested a few changes here and there, but all in all the PR looks great! Thanks for contributing to the lesson!

File: _episodes/03-matplotlib.md (hunk 6 above)
Comment (ldko) [reply _episodes/03-matplotlib.md:20]: I don't think we necessarily need this change, as "First" in the context of all the visualizations made in the episode would be first making the heatmap (needing to import `pyplot` to do that) and second being the mean plot that follows.

File: _episodes/03-matplotlib.md [hunk 7]
```diff
@@ -13,15 +13,28 @@ keypoints:
...
 [heat map]({{ page.root }}/reference.html#heat-map) of our data:
 
+> ## Important Note
```
[the remark this answers is not in the document]
Comment (ldko) [reply _episodes/03-matplotlib.md:23]: I like how using `Prerequisites` as the title would be more specific about the content of the box and also be consistent with what is given as `Prerequisites` at https://swcarpentry.github.io/python-novice-inflammation/.

File: _episodes/03-matplotlib.md (hunk 1 above)
Comment (ldko) [reply _episodes/03-matplotlib.md:25]: Perhaps something like one of the following:

> In the previous episode, [Analyzing Patient Data]({{ page.root }}{% link _episodes/02-numpy.md %}), we imported `numpy` and set the `data` variable. If you have not yet done this, you will need to run the following code before continuing:

or

> If you are not continuing from the previous episode, [Analyzing Patient Data]({{ page.root }}{% link _episodes/02-numpy.md %}), you will need to import the `numpy` library and set the `data` variable:

File: _episodes/03-matplotlib.md (hunk 2 above)
Comment (ldko) [reply _episodes/03-matplotlib.md:33]: I didn't realize `.prereq` was a thing until you mentioned it, and I looked for it :). I think it makes sense to use that for consistency with what is given as `Prerequisites` at https://swcarpentry.github.io/python-novice-inflammation/.

File: _episodes/03-matplotlib.md (hunk 5 above)
Comment (ldko) [reply _episodes/03-matplotlib.md:78]: I don't think we need this change. Since this refers to the value per day across all patients, I don't think we need to add "values"--it also makes the sentence inconsistent with "maximum value" (singular) and a mismatch grammatically with "values" and "seems".

**Review by ldko — COMMENTED** [review]
Thanks very much for following up on this, @GeorgeSavvidis , and adding the box. I have commented further on the changes.

File: _episodes/03-matplotlib.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (maxim-belkin) [reply _episodes/03-matplotlib.md:23]: one more options is `Episode setup`

[the remark this answers is not in the document]
Comment (ldko) [reply _episodes/03-matplotlib.md:23]: that also sounds fine to me
